cli/cli-probe-server/src/main/java/io/telicent/smart/cache/cli/probes/HealthProbeServer.java

```diff
@@ -67,21 +83,60 @@ protected ServerBuilder buildServer() {
      * Runs the server and registers a shutdown hook for cleaning up the server
      */
     public void run() {
-        this.run(false);
+        if (this.future != null) {
+            throw new IllegalStateException("Health Probe Server is already running");
+        }
+        LOGGER.info("Starting Health Probe Server on port {}", this.port);
 
-        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-            if (this.server != null) {
-                this.server.shutdownNow();
+        // Launch this on a background daemon thread
+        this.future = this.executorService.submit(() -> {
+            try {
+                Thread.currentThread().setName("HealthProbeServer");
+            } catch (Throwable e) {
+                // Ignore if unable to set thread name
             }
-        }));
+
+            this.run(true);
+        });
+
+        // Wait briefly to ensure that the server has time to actually start before the caller tries to use it!
+        try {
+            Thread.sleep(250);
```

Comment:
Pedantic: at the very least show it's a long?
```suggestion
            Thread.sleep(250L);
```

Reply:
Done in the tests